amp/shard_tracker.go

```diff
@@ -114,7 +111,11 @@ func (s *ShardTracker) NewShard(pid uint64, last bool) (shards.PaymentShard,
 		}
 
 		s.sharer = sharer
-		child = left.Child(idx)
+
+		// Since splitting the current share added new 32-byte
+		// randomness to the shares, we reset the childIndex.
+		child = left.Child(0)
+		s.childIndex = 0
```

Comment:
maybe it shouldn't be reset, since if you split and merge, then start doing last shards, you get collisions again.